Examples/DotNetCoreBuild/test/TestDirectories/testDirectories.dsc

```diff
@@ -32,11 +32,13 @@ export const explicitOutputsByDifferentPipsAreAllowedInSharedOpaqueDirectory = !
 

```

Comment:
> Suppose that I have a pip P that specifies explicitly that D\f is its output file. It doesn't say anything about D. Another pip Q says that D is its shared output directory. Does the timestamp of the output D\f of P get modified?

Yes, the timestamps of `D/f` do get modified in this case.

> It sounds like all outputs under any shared opaque directories will be scrubbed. Is it the case?

The scrubbing only happens if there is a file on disk that is not part of the pip graph; "real" shared opaque outputs are never part of the pip graph so they get scrubbed (this is exactly how it's always been).  Explicitly declared outputs that are already on disk will not get scrubbed because they are explicitly declared in the pip graph.  So the main purpose of this change is so BuildXL deletes stale outputs that are under shared opaque directories, i.e.,:
  - there once was an explicitly declared output in the pip graph
  - a build was executed and produced a corresponding file on disk (that file gets its timestamps modified with this PR)
  - now that output is removed from the pip graph and a new build is started
    - before this new build is started, the old file on disk is scrubbed because it is not part of the pip graph any more and it does have the special timestamps (which make it look like it was a shared opaque output)

Reply:
Thanks for the explanation.

---
In reply to: [273167983](https://github.com/Microsoft/BuildXL/pull/96#discussion_r273167983) [](ancestors = 273167983)